Terminal.Gui/View/Layout/ViewLayout.cs

```diff
@@ -176,26 +176,31 @@ public virtual Rectangle Bounds
             }
 #endif // DEBUG
 
-            // BUGBUG: I think there's a bug here. This should be && not ||
             if (Margin is null || Border is null || Padding is null)
             {
                 return Rectangle.Empty with { Size = Frame.Size };
             }
 
-            int width = Math.Max (
-                                  0,
-                                  Frame.Size.Width
-                                  - Margin.Thickness.Horizontal
-                                  - Border.Thickness.Horizontal
-                                  - Padding.Thickness.Horizontal
-                                 );
-
-            int height = Math.Max (
-                                   0,
-                                   Frame.Size.Height - Margin.Thickness.Vertical - Border.Thickness.Vertical - Padding.Thickness.Vertical
-                                  );
+            //if (!IsInitialized)
+            //{
+                int width = Math.Max (
+                                      0,
+                                      Frame.Size.Width
```

Comment:
Maybe a good place to use the Offset methods?

One important thing about properties:

A call to set followed immediately by a get should return the original value, in most cases.

When those semantics aren't feasible, the recommendation is typically to use an explicit set method instead of allowing use of the set accessor, or, if still desired, make the set accessor throw an exception instead of messing with input silently.

Reply:
Help my feeble brain work the Math.Max into a use of Offset?

Reply:
This is not offsettting a Rectangle, but changing it's Size.